### ISSUE TITLE
[Union] Hang when using the `all` policy on action or create 

### ISSUE THREAD

**Issue by Roxedus** [rclone/rclone#4534]
#### What is the problem you are having with rclone?
rclone hangs when setting the union remote with the all policy. 

Discussed on the [forums](https://forum.rclone.org/t/union-misbehaving-with-google-drive-and-the-all-policy/18686)

#### What is your rclone version (output from `rclone version`)

`rclone v1.52.3-290-g82975109-beta`
`rclone v1.52.3`

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)
Pop_OS20.04

####  Which cloud storage system are you using? (eg Google Drive)
Jotta and Google Drive, but reproduceable with local directories

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)

```bash
rclone touch union:test
```

#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)

```bash
# rclone touch -vv union:test
2020/08/24 11:58:36 DEBUG : rclone: Version "v1.52.3" starting with parameters ["rclone" "touch" "-vv" "union:test"]
2020/08/24 11:58:36 DEBUG : Using config file from "/config/rclone.conf"
2020/08/24 11:58:37 DEBUG : fs cache: renaming cache item "RostBackup:/BotBox" to be canonical "RostBackup:BotBox"
2020/08/24 11:58:37 DEBUG : fs cache: switching user supplied name "RostBackup:/BotBox" for canonical name "RostBackup:BotBox"
```

**Comment by ncw** [comment]
I replicated this with the config

```
[issue-4535]
type = union
type = union
upstreams = /tmp/issue-4535/1 /tmp/issue-4535/2
create_policy = all
action_policy = all
```

```
mkdir -p /tmp/issue-4535/1 /tmp/issue-4535/2
rclone touch -vv issue-4535:test
```

This appeared to deadlock in the union multiwrite code... Investigating...

**Comment by ncw** [comment]
I managed to fix the above. However there was never an integration test for this and adding one has revealed further problems...

These should be all fixed now. Can you give this a go?

[v1.52.3-340-gfa5d853a-fix-4534-union-all-beta](https://beta.rclone.org/branch/v1.52.3-340-gfa5d853a-fix-4534-union-all-beta/) on branch [fix-4534-union-all](https://github.com/rclone/rclone/tree/fix-4534-union-all) (uploaded in 15-30 mins)

**Comment by Roxedus** [comment]
From my very limited testing, using the config from my forumpost, works with that branch. So it looks good for me

**Comment by ncw** [comment]
I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.53

Thanks for testing :-)
